libs/ledger/src/chain/main_chain.cpp

```diff
@@ -17,30 +17,53 @@
 //------------------------------------------------------------------------------
 
 #include "core/assert.hpp"
-#include "ledger/chain/main_chain.hpp"
-
+#include "core/bloom_filter.hpp"
 #include "core/byte_array/byte_array.hpp"
 #include "core/byte_array/encoders.hpp"
 #include "crypto/hash.hpp"
 #include "crypto/sha256.hpp"
+#include "ledger/chain/main_chain.hpp"
 #include "ledger/chain/transaction_layout_rpc_serializers.hpp"
 #include "network/generics/milli_timer.hpp"
+#include "telemetry/counter.hpp"
+#include "telemetry/registry.hpp"
 
 #include <algorithm>
+#include <cstddef>
+#include <cstdint>
 #include <utility>
+#include <vector>
 
 using fetch::byte_array::ToBase64;
 using fetch::generics::MilliTimer;
 
 namespace fetch {
 namespace ledger {
 
+namespace {
+
+void addBlockToBloomFilter(BloomFilter &bf, Block const &block)
+{
+  for (auto const &slice : block.body.slices)
+  {
+    for (auto const &tx : slice)
+    {
+      bf.Add(tx.digest());
+    }
+  }
+}
+
+}  // namespace
+
 /**
  * Constructs the main chain
  *
  * @param mode Flag to signal which storage mode has been requested
  */
-MainChain::MainChain(Mode mode)
+MainChain::MainChain(std::unique_ptr<BloomFilter> bloom_filter, Mode mode)
+  : bloom_filter_(std::move(bloom_filter))
+  , bloom_filter_false_positive_count_(telemetry::Registry::Instance().CreateCounter(
+        "ledger_main_chain_bloom_filter_false_positive_total"))
```

Comment:
This will need a description to, otherwise that is a warning in prometheus. (I have a PR open to fix the existing ones)

Reply:
Done